controllers/admin/AdminCartRulesController.php

```diff
@@ -640,14 +640,28 @@ public function renderForm()
         $groups = $current_object->getAssociatedRestrictions('group', false, true);
         $shops = $current_object->getAssociatedRestrictions('shop', false, false);
         $cart_rules = $current_object->getAssociatedRestrictions('cart_rule', false, true, 0, $limit);
-        $carriers = $current_object->getAssociatedRestrictions('carrier', true, false);
+        $carriers = $current_object->getAssociatedRestrictions('carrier', true, true);
         foreach ($carriers as &$carriers2) {
-            foreach ($carriers2 as &$carrier) {
+            $prev_id_carrier = 0;
+
+            foreach ($carriers2 as $key => &$carrier) {
+                if ($prev_id_carrier == $carrier['id_carrier']) {
+                    unset($carriers2[$key]);
+
+                    continue;
+                }
+
                 foreach ($carrier as $field => &$value) {
-                    if ($field == 'name' && $value == '0') {
-                        $value = Configuration::get('PS_SHOP_NAME');
+                    if ($field == 'name') {
+                        if ($value == '0') {
+                            $value = Configuration::get('PS_SHOP_NAME');
+                        }
+
+                        $value = $carrier['id_carrier'].' - '.$carrier['name'].' ('.$carrier['delay'].')';
```

Comment:
This would replace the previously assigned `$value` on line 657

Reply:
Fixed!